srml/support/src/dispatch.rs

```diff
@@ -1204,6 +1208,37 @@ macro_rules! __function_to_metadata {
 	}
 }
 
+#[macro_export]
+#[doc(hidden)]
+macro_rules! __check_reserved_fn_name {
+	(deposit_event $( $rest:ident )*) => {
+		$crate::__check_reserved_fn_name!(@compile_error deposit_event);
+	};
+	(on_initialize $( $rest:ident )*) => {
+		$crate::__check_reserved_fn_name!(@compile_error on_initialize);
+	};
+	(on_initialise $( $rest:ident )*) => {
+		$crate::__check_reserved_fn_name!(@compile_error on_initialise);
+	};
+	(on_finalize $( $rest:ident )*) => {
+		$crate::__check_reserved_fn_name!(@compile_error on_finalize);
+	};
+	(on_finalise $( $rest:ident )*) => {
+		$crate::__check_reserved_fn_name!(@compile_error on_finalise);
+	};
+	(offchain_worker $( $rest:ident )*) => {
+		$crate::__check_reserved_fn_name!(@compile_error offchain_worker);
+	};
+	($t:ident $( $rest:ident )*) => {
+		$crate::__check_reserved_fn_name!($( $rest )*);
+	};
+	() => {};
+	(@compile_error $ident:ident) => {
+		compile_error!(concat!("Invalid call: `", stringify!($ident),
```

Comment:
You should mention the expected signature. Otherwise the user does not know what you want from him.

Reply:
ok but at the same time I thought user should refer to decl_module documentation instead of duplicate things here, which may result to outdated doc

Reply:
Then write this into the message. Giving no hint, confuses the user.

Reply:
ok I wrote a better general message, I can still change to a more specific one if this is still wanted.